src/PhpBrew/VariantBuilder.php

```diff
@@ -570,33 +563,26 @@ public function __construct()
          * On ubuntu you need to install libssl-dev
          */
         $this->variants['openssl'] = function (Build $build, $val = null) {
-            if ($val) {
-                return "--with-openssl=$val";
-            }
-
-            if ($prefix = Utils::findIncludePrefix('openssl/opensslv.h')) {
-                return "--with-openssl=$prefix";
-            }
+            $prefix = Utils::findPrefix(array(
+                new BrewPrefixFinder('openssl'),
+                new PkgConfigPrefixFinder('openssl'),
+                new IncludePrefixFinder('openssl/opensslv.h'),
+            ));
 
-            // Special detection and fallback for homebrew openssl
-            // @see https://github.com/phpbrew/phpbrew/issues/607
-            if ($bin = Utils::findBin('brew')) {
-                if ($output = exec_line("$bin --prefix openssl")) {
-                    if (file_exists($output)) {
-                        return "--with-openssl=$output";
-                    }
-                    echo "prefix $output doesn't exist.";
+            if ($build->compareVersion('7.4') < 0) {
+                if ($val) {
+                    return "--with-openssl=$val";
                 }
-            }
 
-            if ($prefix = Utils::getPkgConfigPrefix('openssl')) {
-                return "--with-openssl=$prefix";
-            }
+                if ($prefix !== null) {
+                    return '--with-openssl=' . $prefix;
+                }
+            } else {
+                if ($prefix !== null) {
+                    $build->setPkgConfigPath($prefix.'/lib/pkgconfig');
+                }
 
-            $possiblePrefixes = array('/usr/local/opt/openssl');
-            $foundPrefixes = array_filter($possiblePrefixes, 'file_exists');
-            if (count($foundPrefixes) > 0) {
-                return '--with-openssl=' . $foundPrefixes[0];
+                return '--with-openssl';
```

Comment:
This `return` is redundant. If it's omitted, the result will be the same.

Reply:
I'll remove it.